homeassistant/components/glances/sensor.py

```diff
@@ -80,6 +81,11 @@ def __init__(
 
         self.entity_description = description
         self._attr_name = f"{name} {sensor_name_prefix} {description.name_suffix}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, glances_data.host)},
+            manufacturer="Glances",
+            name=glances_data.name,
```

Comment:
```suggestion
            name=name,
```

Reply:
This wouldn't work properly. In this context `name` is the name of the entity, not the name of the entire Glances server. Basically, device would get named after one of the entities in it - i.e., instead of being called something like "Glances Server" it would be called "Glances Server RAM used"

Reply:
name comes from the entity adding and just looking up into your code name = entry.data[CONF_NAME].
self.name however would be what you describe which is a completely different thing (you can see that you build self.name from name and a few different bits).

Reply:
Lol, you are right! I remembered having this exact issue with `this.name` earlier when implementing `device_info` via property rather than setting an attribute.